### PR TITLE
Get mediaplayer config by mediatype

### PR DESCRIPTION
**Pull request by amoore108**
Related to this comment https://github.com/bbc/simorgh/pull/11383#discussion_r1517471964

Overall changes
======
- Realised that once we create our dedicated route to render the mediaplayer for AMP that the current approach of getting the media config by page-type won't work. We need to get by the media type (e.g `aresMedia` and `clipMedia`) instead. This means it doesn't really matter what page type the media player is on, it depends fully on the block type to get the right config

Code changes
======
- Modify the media player settings builder to check by media type, rather by page type

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/components/MediaLoader/configs/index.ts [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+    'aresMedia',
+  );
+
+  const clipMediaBlock: ClipMediaBlock = filterForBlockType(
+    blocks,
+    'clipMedia',
+  );
+
+  switch (true) {
+    case !!aresMediaBlock:
```
Comment (pvaliani) [inline src/app/components/MediaLoader/configs/index.ts:18]: Is `!!` necessary for each case in this block? Would the switch/case not evaluate it the same without having to explicitly coerce to Boolean?

**Review by pvaliani — APPROVED** [review]
Smol comment but otherwise looks good! nice job.

File: src/app/components/MediaLoader/configs/index.ts (hunk 1 above)
Comment (amoore108) [reply src/app/components/MediaLoader/configs/index.ts:18]: For TypeScript it is necessary as each `case` is evaluated against whats being switched, which is a boolean `true` in this case. The block variables themselves are objects, so as you say it coerces them to boolean to match whats being 'switched'.

Comment (pvaliani) [reply src/app/components/MediaLoader/configs/index.ts:18]: Awesome, thanks for explaining!

**Pull request merged**